ygen/genir_test.go

```diff
@@ -80,11 +86,17 @@ func TestGenerateIR(t *testing.T) {
 						"remote-container": {
 							Name: "RemoteContainer",
 							YANGDetails: YANGNodeDetails{
-								Name:         "remote-container",
-								Defaults:     nil,
-								Module:       "openconfig-simple",
-								Path:         "/openconfig-simple/remote-container",
-								ResolvedPath: "",
+								Name:            "remote-container",
+								Defaults:        nil,
+								BelongingModule: "openconfig-simple",
```

Comment:
See comments elsewhere about these fields. I think that it's quite unclear what these different modules are. It seems to me we have:

 * the module that instantiated the node - i.e., the one that it is within the namespace of.
 * the module that defined the node (where the YANG code exists)

I'm quite unclear what the 'root' module is (see example elsewhere), and think we should seriously consider why this is within the IR.

Reply:
I just updated the doc comment, PTAL: 
https://github.com/openconfig/ygot/blob/927d94dcf31bd2c4126d0f06fb48ab7fe0b2bc7a/ygen/ir.go#L232-L257

The use case here is that ypathgen splits the generated Path struct trees by which module instantiated the root of the tree. This module is the `RootModule`. It happens to also be the first element of `Path` as you identified in [the other PR's comment](https://github.com/openconfig/ygot/pull/657#discussion_r859294556), but if we want to keep the same behaviour we'll need one of them -- currently it seems we don't like either. I personally think it makes sense to split this way in OpenConfig because we don't really care about the namespace (BelongingModule), and definitely not the module containing the text definition (DefiningModule). So, I think we should have this information in the IR.

Reply:
Yep - I think it makes sense to split OpenConfig like this. I think I'm just wanting to try and avoid all the different potential myriad of ways that could be used leaking here. I think if we have the module-qualified path in the IR, we'll have done all the pre-processing, but also be robust to the idea that someone might split packages at the 2nd-level and they will have the right information to do so.

Reply:
Let's discuss this all in the comment above: https://github.com/openconfig/ygot/pull/655#discussion_r860397153